components/AddToCart.js

```diff
@@ -0,0 +1,71 @@
+import cookie from 'js-cookie';
+import React from 'react';
+
+export default function AddToCart(props) {
+  const itemsInCart = cookie.getJSON('cart') || [];
+  console.log('jonas', itemsInCart);
+
+  // useEffect(() => {
+  //   cookie.set('cart', itemsInCart);
+  //   console.log(itemsInCart);
+  // }, [itemsInCart]);
+
+  function MakeCookies() {
+    const product = {
+      name: props.product.name,
+      price: props.total,
+      id: props.product.id,
+      amount: props.pieces,
+      img: props.product.img,
+      info: props.product.info,
+    };
+
+    const itemFilter = itemsInCart.find((item) => item.id === product.id);
+
+    //this part checks if there is an item with the id already present in the array of items in cart
+    if (itemFilter) {
+      const itemsDouble = itemsInCart.map((item) => {
+        if (item.id === props.product.id) {
+          //this condition makes sure the map only adjusts the current item
+          return {
+            ...item,
+            amount: item.amount + props.pieces,
+            price: (item.amount + props.pieces) * item.price,
```

Comment:
Hm, something is off with the math here... Check this out:

1. Go to a product (eg. Creamy Rug) and add a single one to the cart
2. Go to the same product and add a single one to the cart
3. Go to the same product and add a single one to the cart

What should the total be? 3x the single price, right?


<img width="300" src=https://user-images.githubusercontent.com/1935696/87697311-39f91800-c792-11ea-841a-fa522c1ef1c3.png>

<img width="600" src=https://user-images.githubusercontent.com/1935696/87697290-35346400-c792-11ea-94f3-36357f79a522.png>

Reply:
You could write an end-to-end test to verify that this is working :)